app/src/main/java/org/wikipedia/connectivity/NetworkConnectivityReceiver.java

```diff
@@ -30,16 +32,23 @@ public boolean isOnline() {
     @Override public void onReceive(Context context, Intent intent) {
         if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
             NetworkInfo networkInfo = getNetworkInfoFromBroadcast(context, intent);
+            updateOnlineState();
             if (networkInfo != null && networkInfo.isConnected()) {
                 WikipediaApp.getInstance().getBus().post(new NetworkConnectEvent());
             }
-            updateOnlineState();
         }
     }
 
-    private void updateOnlineState() {
+    @VisibleForTesting
+    protected void updateOnlineState() {
+        WikipediaApp app = WikipediaApp.getInstance();
```

Comment:
We should probably not to create a `WikipediaApp` object if we don't use it that often. Please notice that at line `45`, we use the instance too.

Reply:
Done